packages/labs/gen-utils/src/lib/file-utils.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+export interface FileTree {
+  [index: string]: string | FileTree;
+}
+
+/**
+ * Writes a tree of files to an output folder.
+ *
+ * @param outDir Root folder to write files into
+ * @param tree Object with keys either specifying a filename (with value
+ * of string contents) or a folder name (with value of another FileTree object).
+ * As a convenience, filenames may contain slashes, in which case folders
+ * are automatically created.
```

Comment:
What happens if a path is absolute or relative outside the `outDir`? Does this throw?